front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -142,10 +142,13 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
     [projectedTrainsById]
   );
 
-  const trainScheduleUsedForProjection = useMemo(
-    () => (trainIdUsedForProjection ? timetableItemsById.get(trainIdUsedForProjection) : undefined),
-    [trainIdUsedForProjection, timetableItems]
-  );
+  const trainScheduleUsedForProjection = useMemo(() => {
+    if (!timetableItems || timetableItems.length === 0) return undefined;
```

Comment:
this code can be slightly simplified like this 
```suggestion
     if (!timetableItems?.length) return undefined;
```